### DISCUSSION TITLE
How is it done in cl implementation?

### DISCUSSION THREAD

**Discussion by micttyl** [carbon-language/carbon-lang#2147] [General]
In the toolchain, are we going to later use [LLVM Commandline](https://llvm.org/doxygen/CommandLine_8h_source.html) or implement it ourselves?
It is about the argparse-like functionality. This can be questioned from [TODO](https://github.com/carbon-language/carbon-lang/blob/661a3c39074f7539c617743ef4442947803b6954/toolchain/driver/driver.cpp#L74).

**Comment by jonmeow** [comment]
I would lean towards using the LLVM APIs for this (to avoid writing our own), but I think @chandlerc had been hesitant (due to API complexity concerns) -- I think this is something that'll probably be addressed better once we have cause to add more features to the CLI.

**Reply by micttyl** [reply]
driver looks already and almost done without LLVM API, directly to io.

**Reply by micttyl** [reply]
Never mind. I see [TODO](https://github.com/carbon-language/carbon-lang/blob/661a3c39074f7539c617743ef4442947803b6954/toolchain/driver/driver.cpp#L50).
